src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -2203,10 +2281,28 @@ private void tryAddSecurityProvider() {
         });
     }
 
-    // CS-SUPPRESS-SINGLE: RegexpSingleline get Extensions Settings
+    // CS-SUPPRESS-SINGLE: RegexpSingleline get Resource Sharing Extensions
     @Override
     public void loadExtensions(ExtensiblePlugin.ExtensionLoader loader) {
-        // Resource Sharing extensions will be loaded here
+
+        if (settings.getAsBoolean(
+            FeatureConfigConstants.OPENSEARCH_RESOURCE_SHARING_ENABLED,
+            FeatureConfigConstants.OPENSEARCH_RESOURCE_SHARING_ENABLED_DEFAULT
+        )) {
+            Set<String> resourceIndices = new HashSet<>();
+            Map<String, ResourceProvider> resourceProviders = new HashMap<>();
+            Set<ResourceSharingExtension> resourceSharingExtensions = new HashSet<>();
+            for (ResourceSharingExtension extension : loader.loadExtensions(ResourceSharingExtension.class)) {
+                for (ResourceProvider provider : extension.getResourceProviders()) {
+                    resourceProviders.put(provider.resourceIndexName(), provider);
+                    resourceIndices.add(provider.resourceIndexName());
+                }
+                resourceSharingExtensions.add(extension);
+            }
+            resourcePluginInfo.setResourceIndices(resourceIndices);
+            resourcePluginInfo.setResourceProviders(resourceProviders);
+            resourcePluginInfo.setResourceSharingExtensions(resourceSharingExtensions);
```

Comment:
wdyt about only having this setter and then computing everything else within the ResourcePluginInfo class?

Reply:
We have an opportunity to preload the information that will not change, and will allows us to have constant time look-ups instead of computing these from resourceSharingExtensions. I think we should keep it as is.